RFC-0001-torch-function-for-methods.md

```diff
@@ -407,7 +407,114 @@ To implement this proposal requires three main steps:
    arguments that are instances of a type in `types` are processed.
 2. Making sure that all `Tensor` methods except `__new__` and `__init__` go
    through `__torch_function__`.
-3. Add `Tensor.as_subclass` and make `@torch_function_dispatch` public API. 
+3. Add `Tensor.as_subclass` and `@torch_function_dispatch` as public API.
+
+### Implementing only some methods but not others
+One can use the dictionary idiom to only implement some methods but not others.
+A code example follows:
+
+```python
+HANDLED_FUNCTIONS = {}
+
+def implements(func):
+    def inner(implementation):
+        HANDLED_FUNCTIONS[func] = implementation
+        return implementation
+
+@implements(torch.add)
+def my_add(self, other):
+    ...
+
+class TensorLike:
+    @classmethod
+    def __torch_function__(cls, func, types, args, kwargs):
+        implementation = HANDLED_FUNCTIONS.get(func, None)
+        if implementation is None:
+            return NotImplemented
+        
+        return implementation(*args, **kwargs)
+```
+
+For subclasses, one can also choose to use the fallback implementation if
+a specialized implementation isn't available using `super`, as shown below.
+
+```python
+class SubTensor(torch.Tensor):
+    @classmethod
+    def __torch_function__(cls, func, types, args, kwargs):
+        implementation = HANDLED_FUNCTIONS.get(func, None)
+        if implementation is None:
+            return super().__torch_function__(
+                func, types, args, kwargs
+            )
+        
+        return implementation(*args, **kwargs)
+```
+
+A call to `super().__torch_function__` can also be used to call the fallback
+implementation within any other function.
+
+The examples we have seen here actually specify what we anticipate will be two
+common patterns of using `__torch_function__`: `LoggingTensor` is an example
+of a global hook, and the two examples above show a way to achieve specialised
+implementations of particular functions.
+
+### Functions vs Methods vs Properties
+Both functions and methods/properties on `torch.Tensor` will be possible arguments to
+`__torch_function__`. These are different in subtle but important ways, and
+in some cases it is required to handle them differently. For instance,
+`torch.Tensor` methods/properties have the following properties:
+
+1. They can only accept `torch.Tensor` instances as the first argument.
+2. They *may or may not* have a `__module__` defined.
```

Comment:
This kind of comes out of the blue; maybe this would be better in a FAQy or appendix section